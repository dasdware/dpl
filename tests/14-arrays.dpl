var ns: [Number] := [10, 20, 30, 40, 50];
print("${ns.length()}\n");

for (var i in 0..(ns.length()-1)) {
   print("#${i+1}: ${ns[i]}\n");
};

var empty := [];
print("${empty.length()}\n");

var strings: [String] := empty;
print("${strings.length()}\n");

ns := [0, ..ns, (ns.length() + 1) * 10];
for (var i in 0..(ns.length()-1)) {
    print("#${i+1}: ${ns[i]}\n");
};

for (var i in ns) {
    print("#${i}\n");
};

#function add(ns1: [Number], ns2: [Number])
#    := [..ns1, ..ns2];

#for (var n in (numbers + [5]))
#  print("${n}\n");