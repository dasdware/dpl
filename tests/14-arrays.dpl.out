5
#1: 10
#2: 20
#3: 30
#4: 40
#5: 50
0
0
#1: 0
#2: 10
#3: 20
#4: 30
#5: 40
#6: 50
#7: 60
